app/controllers/api/base_controller/generic.rb

```diff
@@ -19,6 +19,7 @@ def add_resource(type, _id, data)
             data.delete(sc.to_s)
           end
         end
+        validate_type(klass, data['type']) if data['type']
```

Comment:
are we saying data['type'] is always the class/subclass in all collections ?  or is this only for specific collections ?  If so, this might bite us later and we may want to implement in the relevant collection implementation with a super.

Reply:
To be honest I don't think I'd bother here. This hash is just passed raw in to the model in the next line and it's a fair to assume that passing `type` as an attribute to any ActiveRecord model is going to be problematic in some way and need adjustments if in the future there's some reason you want to be able to pass a `type` that isn't referring to a polymorphic type.

Reply:
👍  in general too (future fine tuning as needed), I just didn't know offhand if we had such a beast today with all the collections we support. Maybe a quick perusal/sanity check.

Reply:
Yup, looked around already; my glance didn't yield anything that wasn't an overloaded STI class.

Reply:
Thanks !!  👍